### PR TITLE
PKI read legacy CA bundle prior to migration

### PR DESCRIPTION
**Pull request by stevendpclark**
When the new Vault version starts up and becomes the leader node on a secondary performance cluster it should be able to still support certificate issue/revoke and list commands even when the storage was not migrated.

### Known limitations

The new api on the secondary does not return the proper information in regards to issuers and keys as we don't fallback for those api calls.

### Testing performed

- Started up a primary and secondary performance clusters on 1.10 setting up a full CA within two separate mounts. 
- Upgraded the secondary performance cluster and verified that I could issue a new certificate from it as well as revoking a cert. 
- Upgraded the primary and verified that the new issuers api started working without restarting Vault on the secondary performance cluster.

### CODE REVIEW

File: builtin/logical/pki/backend.go
Comment (cipherboy) [inline builtin/logical/pki/backend.go:308]: Since we merged the CRL stuff, a call to `buildCRLs` should suffice here. :-)

File: builtin/logical/pki/storage_migrations.go
Comment (cipherboy) [inline builtin/logical/pki/storage_migrations.go:42]: If  this has been deleted (or never created), does it mean we'll err out here?

File: builtin/logical/pki/backend.go
Comment (cipherboy) [inline builtin/logical/pki/backend.go:286]: And related to the above, wondering if that means the migration status will be incorrect here? Namely, the `if !info.isRequired {...}` check later might need to be moved up?

File: builtin/logical/pki/storage_migrations.go
Comment (cipherboy) [reply builtin/logical/pki/storage_migrations.go:42]: Ah, I think we don't err. Only on other storage issues.

File: builtin/logical/pki/cert_util.go
Comment (cipherboy) [inline builtin/logical/pki/cert_util.go:142]: Not to nitpick, but doesn't this turn this into an `InternalError` versus a `UserError`? Not sure how much the distinction matters, but seems like we'd want to preserve that.

File: builtin/logical/pki/path_intermediate.go
Comment (cipherboy) [inline builtin/logical/pki/path_intermediate.go:51]: nit: `s/intermediary/intermediate/g` perhaps, for consistent language?

**Review by cipherboy — APPROVED** [review]
Largely looks good, some minor nits I think. 

Will this PR handle updating the `buildCRL` path (now that it has merged) or should I do that after this merges? 

Also, not sure if the complete "new APIs should error" will be done here now or in a separate PR :-)

File: builtin/logical/pki/cert_util.go
Comment (stevendpclark) [reply builtin/logical/pki/cert_util.go:142]: Yup good catch, that was a flub on my part.

File: builtin/logical/pki/storage_migrations.go
Comment (stevendpclark) [reply builtin/logical/pki/storage_migrations.go:42]: Correct, but this comment led me to a missing nil check within fetchCAInfo so thanks 👍

File: builtin/logical/pki/backend.go
Comment (stevendpclark) [reply builtin/logical/pki/backend.go:286]: If you are referring to the empty/non-existant ca-bundle then that won't be a problem. Please let me know if I have misunderstood your concern.

Comment (cipherboy) [reply builtin/logical/pki/backend.go:286]: Yeah, it was about the above comment, but I see that's wrong, feel free to ignore this one :-)

File: builtin/logical/pki/backend.go
Comment (stevendpclark) [reply builtin/logical/pki/backend.go:309]: Will defer to another PR, since we don't have a request object here this isn't as trivial as calling buildCRLs.

File: builtin/logical/pki/backend.go
Comment (cipherboy) [inline builtin/logical/pki/backend.go:294]: nit: perhaps drop the status?

**Pull request merged**
